state/status.go

```diff
@@ -280,7 +280,14 @@ func setStatus(db Database, params setStatusParams) (err error) {
 		StatusData: utils.EscapeKeys(params.rawData),
 		Updated:    params.updated.UnixNano(),
 	}
-	probablyUpdateStatusHistory(db, params.globalKey, doc)
+
+	newStatus, historyErr := probablyUpdateStatusHistory(db, params.globalKey, doc)
+	if !newStatus && historyErr == nil {
+		// If this status is not new (i.e. it is exactly the same as
+		// our last status), there is no need to update the record.
+		// Update here will only reset the 'Since' field.
+		return
```

Comment:
Please use `return err`. If we aren't going to care what the error is, why return it?

Reply:
done